cli/cli-improvements.md

```diff
@@ -0,0 +1,315 @@
+* we will establish the convention that by default, an application ```<app-name>``` to be deployed using environment ```<env-name>``` will be a under the resource-group ```<app-name>-<env-name>```
+
+The workspace entry created as part of rad init will be as below: 
+
+**Current config.yaml**
+
+```
+workspaces:
+    default: default
+    items:
+        default:
+            connection:
+                context: radius-agent-aks
+                kind: kubernetes
+            environment: /planes/radius/local/resourceGroups/default/providers/Applications.Core/environments/default
+            scope: /planes/radius/local/resourceGroups/default
+```
+
+**Proposed config.yaml**
+
+```
+workspaces:
+    default: default
+    items:
+        default:
+            connection:
+                context: radius-agent-aks
+                kind: kubernetes
+            environment: /planes/radius/local/resourceGroups/default/providers/Applications.Core/environments/default
+```
+
+All CLI commands will be impacted due to this change.
+
+
+### Example
+
+Consider a user wants to deploy two instances of `cool-app` application, into `production` and `qa` environments. 
+
+#### How can the user do it today
+
+```
+1. rad init
+2. rad group create qa
+3. rad group switch qa
+4. rad env create qa 
+5. rad env switch qa
+6. rad deploy cool-app
+
+7. rad group create production  #neccessary, since creating and using a new env alone wont change resource ids and there might be different rbac requirements
+8. rad group switch production
+9. rad env create production 
+10. rad env switch production
+11. rad deploy cool-app
+```
+
+#### How it would change
+
+```
+1. rad init # ability to enter "production" as default env's name, which would create this env in "all-environments" resource group. 
+```
+resulting config.yaml
+```
+workspaces:
+    default: default
+    items:
+        default:
+            connection:
+                context: radius-agent-aks
+                kind: kubernetes
+            environment: /planes/radius/local/resourceGroups/all-environments/providers/Applications.Core/environments/production
+```
+```
+
+2. rad env create qa -g all-environments
+
+3. rad deploy cool-app
+   which env would you like to deploy to? [prefilled to /planes/radius/local/resourcegroups/all-environments/production] 
+   `cool-app` being deployed using environment production in resource group cool-app-production. 
+   Use rad deploy <app-name> -g <group-name> -e <env-name> 
+   to change the behavior. Proceed? [Y/n]
+   :
+
+4. rad deploy cool-app
+   which env would you like to deploy to? [prefilled to production but set to /planes/radius/local/resourcegroups/all-environments/providers/applications.core/environments/qa] 
+   `cool-app` being deployed using environment qa in resource group cool-app-qa.
+   Use rad deploy <app-name> -g <group-name> -e <env-name> 
+   to change the behavior. Proceed? [Y/n] 
+   :
```

Comment:
Reminder: we want to be really clear and specific about which commands are interactive and not. Ideally we have few interactive commands.